content/issues/2023-05-28-mariadb.md

```diff
@@ -0,0 +1,43 @@
+---
+title: 'Known Issue: MariaDB Check and Repair'
+date: '2023-05-28 12:00:00Z'
+severity: notice
+resolved: false
+affected:
+  - 'Known Issues'
+  - 'mariadb'
+section: 'issue'
+---
+There is a known issue with MariaDB when updating the container image and a database upgrade is required on user databases that are not in a healthy state. This can prevent MariaDB from starting.
+
+MariaDB version updates (ex: `10.6.x` to `10.11.x`) can happen when updating the container image. Upgrading user databases is done manually by running a command.
+
+When the container initializes, if `MYSQL_ROOT_PASSWORD` is set an upgrade check will run. If an upgrade is required the log will indicate the need to run:
+
+```shell
+mariadb-upgrade -u root -p<PASSWORD>
+```
+
+If user databases are not in a healthy state (sometimes caused by a failed upgrade), it may be remedied by running:
+
+```shell
+mariadb-check -u root -p<PASSWORD> -c -A # check all databases for errors
+mariadb-check -u root -p<PASSWORD> -r -A # repair all databases
+mariadb-check -u root -p<PASSWORD> -a -A # analyze all databases
+mariadb-check -u root -p<PASSWORD> -o -A # optimize all databases
+```
+
+After running the above commands, you may need to run the upgrade command again.
+
+If you have recently updated your container image and your container logs show repeating entries of:
+
+```log
+Caught SIGTERM signal!
+```
+
+- Roll back the container image to a previous tag (ex: `lscr.io/linuxserver/mariadb:10.6.13` assuming this was the version you were running when things were last working)
+- Run the check, repair, analyze, and optimize commands above, watch for any issues
+- Update the container image to the new tag (ex: `lscr.io/linuxserver/mariadb:10.11.3` or `lscr.io/linuxserver/mariadb:latest`)
+- Run the upgrade command above
+
+This should get MariaDB running again with your user databases in a healthy state.
```

Comment:
I would add a note at the end to the effect of
```
If running MariaDB databases in production, we always recommend pinning the container to a specific version tag, such as `version-10.6.13-r0`, and planning any upgrades so that you can test and monitor them.
```

Reply:
Resolved in 5966df5b9120f16eeee8ba1379e7ec691c1a6197